lib/matplotlib/axes/_subplots.py

```diff
@@ -157,33 +157,33 @@ def label_outer(self):
             self.get_yaxis().get_offset_text().set_visible(False)
             self.set_ylabel("")
 
-    def _make_twin_axes(self, *kl, **kwargs):
+    def _make_twin_axes(self, *args, **kwargs):
         """
         Make a twinx axes of self. This is used for twinx and twiny.
         """
-        from matplotlib.projections import process_projection_requirements
         if 'sharex' in kwargs and 'sharey' in kwargs:
             # The following line is added in v2.2 to avoid breaking Seaborn,
             # which currently uses this internal API.
             if kwargs["sharex"] is not self and kwargs["sharey"] is not self:
-                raise ValueError("Twinned Axes may share only one axis.")
-        kl = (self.get_subplotspec(),) + kl
-        projection_class, kwargs, key = process_projection_requirements(
-            self.figure, *kl, **kwargs)
-
-        ax2 = subplot_class_factory(projection_class)(self.figure,
-                                                      *kl, **kwargs)
-        self.figure.add_subplot(ax2)
+                raise ValueError("Twinned Axes may share only one axis")
+        # The dance here with label is to force add_subplot() to create a new
+        # Axes (by passing in a label never seen before).  Note that this does
+        # not affect plot reactivation by subplot() as twin axes can never be
+        # reactivated by subplot().
+        sentinel = object()
+        real_label = kwargs.pop("label", sentinel)
+        twin = self.figure.add_subplot(
+            self.get_subplotspec(), *args, label=sentinel, **kwargs)
```

Comment:
Is it safe to set *label* to an `object` instance? According to the docstring it must be a `str`. Even if it works now, it might cause problems with future changes. Maybe use something like `str(uuid.uuid4())` instead?

Reply:
done